### PR TITLE
SHE: autogen secrets containing etcd tls assets

### PR DESCRIPTION
**Pull request by hongchaodeng**
Step 2 of https://github.com/kubernetes-incubator/bootkube/issues/582

### CODE REVIEW

File: pkg/asset/asset.go [hunk 1]
```diff
@@ -56,6 +56,9 @@ const (
 	AssetPathSystemNamespace                = "manifests/kube-system-ns.yaml"
 	AssetPathCheckpointer                   = "manifests/pod-checkpointer.yaml"
 	AssetPathEtcdOperator                   = "manifests/etcd-operator.yaml"
+	AssetPathSelfHostedEtcdMemberPeerSecret = "manifests/etcd-member-peer-tls.yaml"
+	AssetPathSelfHostedEtcdMemberCliSecret  = "manifests/etcd-member-client-tls.yaml"
+	AssetPathSelfHostedOperatorEtcdSecret   = "manifests/operator-etcd-client-tls.yaml"
```
Comment (dghubble) [inline pkg/asset/asset.go]: Can you clarify the intended use of the MemberPeer, MemberCli, and Operator certificates? This came up previously when client certs were almost used between etcd peers.

Comment (dghubble) [inline pkg/asset/asset.go]: Disregard, moving to the issue discussion.

Comment (diegs) [inline pkg/asset/asset.go]: EtcdOperator, etcd-operator, etc?

File: pkg/asset/k8s.go [hunk 2]
```diff
@@ -96,6 +99,42 @@ func newKubeConfigAsset(assets Assets, conf Config) (Asset, error) {
...
+		AssetPathSelfHostedEtcdMemberClientCA,
+		AssetPathSelfHostedEtcdMemberClientCert,
+		AssetPathSelfHostedEtcdMemberClientKey,
+	}, assets)
+	if err != nil {
+		return nil, err
+	}
+	res = append(res, Asset{Name: AssetPathSelfHostedEtcdMemberCliSecret, Data: secretYAML})
+
+	secretYAML, err = secretFromAssets(secretOperatorEtcd, secretNamespace, []string{
```
Comment (diegs) [inline pkg/asset/k8s.go]: EtcdOperator etc.?

File: pkg/asset/asset.go (hunk 1 above)
Comment (hongchaodeng) [inline pkg/asset/asset.go]: Sure.
We provide a docs at https://github.com/coreos/etcd-operator/blob/master/doc/user/cluster_tls.md .
Let me know if you don't understand anything in the docs.

> This came up previously when client certs were almost used between etcd peers.

That's the whole point of operator -- expertise written into the code.

Comment (hongchaodeng) [inline pkg/asset/asset.go]: Just to make sure we are consistent, this is (Etcd Operator)'s (Etcd Client)'s secret.

But change it to EtcdOperatorSecret is fine too.

**Comment by hongchaodeng** [comment]
fixed

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Pull request merged**